### PR TITLE
feat(feedback): Implement feedback feature! (#28)

### LINKED ISSUE #28: Report bug/request feature
As a user, I want to be able to report to the developers when a bug occurs, have a suggestion or request a feature

### Dev notes

Use [BetterFeedback](https://pub.dev/packages/feedback) package, then create a GitHub issues

### PR DESCRIPTION
**Pull request by apomalyn**
## 📖 Description

Implementation of feedback option. This option can be found under the _General_ section of the settings.
It will open a bottom sheet with the option to create a **feature request** or **report a bug**. It will then ask the user to give more information on the feedback he/she wants to offer, and finally, create a GitHub issue on this repository

The user also has the option to take a screenshot of the application using the [Feedback](https://pub.dev/packages/feedback).

#### Rules applied on the feedback form

- **email**: optional on both cases, when the user profile is added, it will become a checkbox 'send as Anonyme'
- **description**: only required for `feature request`
- **steps to reproduce**: only required for `bug report`
- **screenshot**: only available for `bug report` but optional.

#### Other changes

- Introductions of `DialogService`, a wrapper allowing use to create dialogs without the context
- Enhancement of `InfoService` to have device information like the device model
- Adding a simple extension to get a `MaterialColor` from a `Color`

### ⁉️ Related Issues

closes #28 

### 🖼️ Screenshots:

<details>
    <summary>Demonstration 'Feature request' on iOS, the issue created --> #89 </summary> 

https://github.com/RoadTripMoustache/kana_to_kanji/assets/22211097/84b213e6-73cd-4b7a-a44f-cb154f161604

</details>

<details>
    <summary>Demonstration 'Report bug' on iOS, the issue created --> #90 </summary> 

https://github.com/RoadTripMoustache/kana_to_kanji/assets/22211097/1aa402c1-6199-4b12-a7c2-27b0e95aa9ec

</details>

The Android build looks the same, except for one detail. On the screenshot screen, the _Report_ button is more prominent, I try to find a solution without success.

### 🧪 How to test the change?

1. If you are testing locally, make sure to update your `.env`, run `flutter clean` and regenerate the files
2. Go to the _Settings_ view
3. Click on _Give feedback_ button
4. See changes

### ☑️ Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] I have added/updated tests.
- [x] Make sure to add either one of the following labels: `version: Major`,`version: Minor` or `version: Patch`.

### CODE REVIEW

File: lib/src/core/utils/extensions.dart [hunk 1]
```diff
@@ -1,7 +1,23 @@
...
 import 'package:kana_to_kanji/src/core/constants/alphabets.dart';
 
 extension AlphabetsExtension on Alphabets {
   bool isKana() {
     return this == Alphabets.hiragana || this == Alphabets.katakana;
   }
 }
+
+extension ColorExtension on Color {
+  MaterialColor toMaterialColor() => MaterialColor(value, {
```
Comment (Adaendra) [inline lib/src/core/utils/extensions.dart:11]: A quoi ça correspond?

File: lib/src/feedback/feedback_view_model.dart [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+import 'package:kana_to_kanji/src/feedback/utils/build_issue_helper.dart';
+import 'package:kana_to_kanji/src/feedback/widgets/feedback_success_dialog.dart';
+import 'package:kana_to_kanji/src/locator.dart';
+import 'package:stacked/stacked.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:kana_to_kanji/src/feedback/constants/feedback_form_fields.dart';
+import 'package:kana_to_kanji/src/feedback/constants/feedback_type.dart';
+import 'package:image/image.dart' as image;
+
+const _kScreenshotWidth = 300;
```
Comment (Adaendra) [inline lib/src/feedback/feedback_view_model.dart:19]: Ca manque d'un petit commentaire pour expliquer à quoi ça correspond

File: lib/src/core/utils/extensions.dart (hunk 1 above)
Comment (apomalyn) [reply lib/src/core/utils/extensions.dart:11]: Avec Material 3, il y a plein de nouvelle fonction qui change une couleur automatiquement en fonction du background, foreground, couleur voisines, etc...Du coup ils ont créé `MaterialColor` et `MaterialColorState`. Depuis, certaines fonction n'accepte pu de `Color` mais des `MaterialColor`, cette extension de `Color` va nous permettre de construire on the fly une `MaterialColor`.

Alors techniquement je pourrais le retirer puisque je l'utilisais et ne l'utilise plus, par contre ca reste utile c'est pour ca que je l'ai laissé. Je peux la retirer si tu prefere :D

Comment (Adaendra) [reply lib/src/core/utils/extensions.dart:11]: On peut le laisser, mais il faudrait laisser un commentaire pour expliquer à quoi ça correspond et potentiellement mettre un lien d'une doc externe si jamais y'en a une

Comment (apomalyn) [reply lib/src/core/utils/extensions.dart]: Je l'ai retirer, on verra si on en a besoin plus tard

**Pull request merged**